hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -225,9 +223,18 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 	switch nodePool.Spec.Platform.Type {
 	case hyperv1.AWSPlatform:
 		md := machineDeployment(nodePool, hcluster.Spec.InfraID)
-		ami, err := r.ImageProvider.Image(hcluster)
-		if err != nil {
-			return ctrl.Result{}, fmt.Errorf("failed to obtain AMI: %w", err)
+		var ami string
+		if arch, foundArch := releaseImage.StreamMetadata.Architectures["x86_64"]; !foundArch {
+			return ctrl.Result{}, fmt.Errorf("couldn't find OS metadata for architecture %q", "x64_64")
+		} else {
```

Comment:
can we signal all this new errors in a nodePool condition as well?

Reply:
That seems like it would be useful

Reply:
I incorporated a new `AMIDiscoveryFailed` condition to surface this failure